### DISCUSSION TITLE
How to use the Security Addon

### DISCUSSION THREAD

**Discussion by schroe-ja** [juhaku/utoipa#786] [Q&A]
Hi,

I added the `SecurityAddon` trait but I still don't have to add an API key. How can I enforce it for every endpoint?
I used the `todo-actix` example to implement the Swagger UI.

**Comment by juhaku** [comment]
## Add to each path separately

https://github.com/juhaku/utoipa/blob/67f04b3876dab30790bb4ce256e9c7f41226a041/examples/todo-actix/src/main.rs#L50-L58
In the above code at the line 54 https://github.com/juhaku/utoipa/blob/67f04b3876dab30790bb4ce256e9c7f41226a041/examples/todo-actix/src/main.rs#L54
Must match to the `api_key` value on a `#[utoipa::path(..)]`
https://github.com/juhaku/utoipa/blob/67f04b3876dab30790bb4ce256e9c7f41226a041/examples/todo-actix/src/todo.rs#L139-L155

## Add to whole API via OpenApi `security` field

However you can add the `api_key` requirement manually for the whole API by manually adding it to: https://github.com/juhaku/utoipa/blob/67f04b3876dab30790bb4ce256e9c7f41226a041/utoipa/src/openapi.rs#L107

You can also use the `#[derive(OpenApi)]` macro to add the security requirement for the whole API as defined here https://docs.rs/utoipa/latest/utoipa/openapi/security/struct.SecurityRequirement.html#examples-1
